dream-web/view/github.ml

```diff
@@ -0,0 +1,399 @@
+  in
+  statuses (List.fold_left render_job [] jobs)
+
+let list_orgs ~orgs = Template.instance @@ orgs_v ~orgs
+let list_repos ~org ~repos = Template.instance @@ repos_v ~org ~repos
+
+let list_refs ~org ~repo ~refs =
+  Template.instance
+    [
+      breadcrumbs [ ("github", "github"); (org, org) ] repo;
+      refs_v ~org ~repo ~refs;
+    ]
+
+let cancel_success_message success =
+  let open Tyxml.Html in
+  let format_job_info ji =
+    li [ span [ txt @@ Fmt.str "Cancelling job: %s" ji.Client.variant ] ]
+  in
+  match success with
+  | [] -> div [ span [ txt @@ Fmt.str "No jobs were cancelled." ] ]
+  | success -> ul (List.map format_job_info success)
+
+let cancel_fail_message failed =
+  let open Tyxml.Html in
+  match failed with
+  | n when n <= 0 -> div []
+  | 1 ->
+      div
+        [
+          span
+            [
+              txt
+              @@ Fmt.str
+                   "1 job could not be cancelled. Check logs for more detail.";
+            ];
+        ]
+  | n ->
+      div
+        [
+          span
+            [
+              txt
+              @@ Fmt.str
+                   "%d jobs could not be cancelled. Check logs for more detail."
+                   n;
+            ];
+        ]
+
+let rebuild_success_message success =
+  let open Tyxml.Html in
+  let format_job_info ji =
+    li [ span [ txt @@ Fmt.str "Rebuilding job: %s" ji.Client.variant ] ]
+  in
+  match success with
+  | [] -> div [ span [ txt @@ Fmt.str "No jobs were rebuilt." ] ]
+  | success -> ul (List.map format_job_info success)
+
+let rebuild_fail_message failed =
+  let open Tyxml.Html in
+  match failed with
+  | n when n <= 0 -> div []
+  | 1 ->
+      div
+        [
+          span
+            [
+              txt
+              @@ Fmt.str
+                   "1 job could not be rebuilt. Check logs for more detail.";
+            ];
+        ]
+  | n ->
+      div
+        [
+          span
+            [
+              txt
+              @@ Fmt.str
+                   "%d jobs could not be rebuilt. Check logs for more detail." n;
+            ];
+        ]
+
+let return_link ~org ~repo ~hash =
+  let open Tyxml.Html in
+  let uri = commit_url ~org ~repo hash in
+  a ~a:[ a_href uri ] [ txt @@ Fmt.str "Return to %s" (short_hash hash) ]
+
+(* TODO: Clean up so that success and fail messages appear in flash messages and we do a redirect
+instead of providing a return link *)
+let list_steps ~org ~repo ~refs ~hash ~jobs
+    ?(success_msg =
+      let open Tyxml.Html in
+      div [])
+    ?(fail_msg =
+      let open Tyxml.Html in
+      div [])
+    ?(return_link =
+      let open Tyxml.Html in
+      div []) ?(flash_messages = []) ~csrf_token () =
+  let open Tyxml.Html in
+  let can_cancel =
+    List.fold_left
+      (fun accum job_info ->
+        accum
+        ||
+        match job_info.Client.outcome with
+        | Active | NotStarted -> true
+        | Aborted | Failed _ | Passed | Undefined _ -> false)
+      false jobs
```

Comment:
Maybe this can be written with a simplified version with:
```suggestion
     let check { job_info.Client.outcome; _ } with
        match outcome with
        | Active | NotStarted -> true
        | Aborted | Failed _ | Passed | Undefined _ -> false)
     in
    List.exists check jobs
```